streamer/pulp/streamer/server.py

```diff
@@ -52,6 +52,36 @@ class DownloadListener(AggregatingEventListener):
     Nectar download listener.
     """
 
+    def __init__(self, streamer, request):
+        """
+        :param streamer: The streamer.
+        :type  streamer: Streamer
+        :param request: An HTTP request.
```

Comment:
As a general comment, I'm reminded that every time I look at the streamer and see a "request" or "response", I have to think carefully about where it came from. Since the streamer handles requests and makes new requests, and of course thus it receives responses and uses those to generate new responses, it gets confusing.

I wish the doc blocks were consistently explicit about whether the streamer is acting as the client or server with respect to each request and response. It would be helpful to clarify that here.